pages/categories/[slug].js

```diff
@@ -1,25 +1,66 @@
 import { useRouter } from "next/router";
 import Layout from "~/components/Layout/Layout";
 import useSWR from "swr";
+import Header from "~/components/Header";
+import SubCatCard from "~/components/SubCatCard";
+import headerImgCat from "../../public/images/headerImgCat.jpeg";
+import titleConvertor from "~/lib/titleConvertor";
+
 const Post = () => {
   const router = useRouter();
   const { slug } = router.query;
-  const { data, error } = useSWR(slug ? "/api/categories/" + slug : undefined);
+  const { data: issuesCategories, error } = useSWR("/api/issues");
 
-  if (!data && !error) {
+  if (!issuesCategories && !error) {
     return <div>Loading...</div>;
   }
-  console.log(data, error);
 
-  if (!data) {
+  if (!issuesCategories) {
     return <Layout>Not found.</Layout>;
   }
 
   return (
-    <Layout title="issues">
-      <h1 className="text-red-800 bg-yellow-300 font-bold text-5xl p-32">
-        Hey hey hey from {data.data.name} page!
-      </h1>
+    <Layout title={slug}>
+      <Header
+        backgroundColor="white"
+        imgSrc={headerImgCat}
+        imgAlt="Woman's hand"
+        title={titleConvertor(slug)}
+        caption={
+          "Miricyl helps you find key, informative resources to guide you on your mental health journey"
+        }
+      />
+      <div className="flex flex-col md:flex-row md:flex-wrap justify-end items-center md:justify-between my-12 sm:my-32 sm:mx-12">
+        {issuesCategories &&
+          issuesCategories.data
+            .filter((issue) => issue.slug === slug)
+            .map(({ id, category_id, name }) => (
+              <SubCatCard
+                key={id}
+                backgroundColor1={
+                  category_id === 1
+                    ? "yellow"
+                    : category_id === 2
+                    ? "green"
+                    : category_id === 3
+                    ? "indigo"
+                    : "indigo"
```

Comment:
This seems a little clunky. Why not try something like `backgroundColor1={getColorForCategoryId(category_id)}` and then you could write a short function that returns the colour string depending on what argument it gets passed?

I think that would make it easier to read.